iDiff/differs/history.go

```diff
@@ -8,31 +8,45 @@ import (
 	"strings"
 
 	"github.com/GoogleCloudPlatform/runtimes-common/iDiff/utils"
-
+	"github.com/docker/docker/api/types/image"
 	"github.com/docker/docker/client"
+	"github.com/golang/glog"
 	"golang.org/x/net/context"
 )
 
 // History compares the Docker history for each image.
-func History(img1, img2 string, json bool) (string, error) {
-	return getHistoryDiff(img1, img2, json)
+func History(img1, img2 string, json bool, eng bool) (string, error) {
+	return getHistoryDiff(img1, img2, json, eng)
 }
 
-func getHistoryList(image string) ([]string, error) {
-	ctx := context.Background()
-	cli, err := client.NewEnvClient()
+func getHistoryList(img string, eng bool) ([]string, error) {
+	validDocker, err := utils.ValidDockerVersion(eng)
 	if err != nil {
 		return []string{}, err
 	}
-	history, err := cli.ImageHistory(ctx, image)
-	if err != nil {
-		return []string{}, err
+	var history []image.HistoryResponseItem
+	if validDocker {
+		ctx := context.Background()
+		cli, err := client.NewEnvClient()
```

Comment:
The real issue here is that you end up setting up a client with a much higher client API version than your server if you don't have `DOCKER_API_VERSION` set.
Basically I think you're trying to avoid the code in `client.NewEnvClient()`

```golang
version := os.Getenv("DOCKER_API_VERSION")
if version == "" {
  version = api.DefaultVersion //1.31
}
```

I think you want this convenience method to set the client version appropriately.
```golang
// NegotiateAPIVersion updates the version string associated with this
// instance of the Client to match the latest version the server supports
func (cli *Client) NegotiateAPIVersion(ctx context.Context) {
```

https://github.com/moby/moby/blob/ff4f700f74450018f36d014f3cde0ff1b9c17fb3/client/client.go#L244-L249